### PR TITLE
Closes #2677: Fix broken Google Tag config upgrade.

### LINKED ISSUE #2677: Google Tag manager disabled for anonymous users after upgrading to 2.7.x
## Problem/Motivation
Google Tag containers are no longer loaded for anonymous users on Quickstart sites after upgrading to 2.7.x.

### Describe the bug
Drupal.org issue submitted: [GoogleTagUpgradeManager not mapping negate values correctly](https://www.drupal.org/project/google_tag/issues/3382757)

### To Reproduce
- Upgrade existing 2.6.x site to 2.7.x
- Notice that GTM containers no longer loaded for anonymous users

## Proposed resolution
- Add patch submitted to Drupal.org to `composer.json` for Google Tag module
- Consider adding DB update to Quickstart to update cleanup configs messed up by module DB update

### Expected behavior
Upgrading from Quickstart 2.6.x to 2.7.x should not break GTM configuration

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
Applies patch to google_tag module to fix issue with module-provided DB update that broke Google Tag container configs on sites upgrading from AZQS 2.6.x to 2.7.x.

https://www.drupal.org/project/google_tag/issues/3382757

### To Do

- [x] Update exported Google tag manager config included in Quickstart with effects of updated DB update 
- [ ] Still need to decide whether to add a Quickstart-specific DB update to fix configs screwed up by the google_tag module update

## Related issues

Closes #2677

## How to test

- Create a Quickstart 2.6.x site
- Upgrade to Quickstart 2.7.x using this PR branch
- Verify that GTM config works the same way

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [x] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by trackleft** [comment]
Tested this and it appears to work.

**Comment by joeparsons** [comment]
Per 2023-08-23 meeting discussion we are going to hold off on merging this until at least 2023-08-25 (Friday meeting) so that we can see how the module maintainers respond before we pre-emptively make any DB updates, etc.

**Review by trackleft — APPROVED** [review]

**Comment by joeparsons** [comment]
Update:  The module maintainers have committed our patch!
https://www.drupal.org/project/google_tag/issues/3382757#comment-15203863

**Review by tadean — APPROVED** [review]

**Pull request merged**
